csrc/fusion_segmenter.cpp

```diff
@@ -2523,7 +2523,7 @@ class FusionSegmentGuard : public NonCopyable {
     }
 
     for (auto old_out : old_outputs_) {
-      fusion_->addOutput(old_out);
+      fusion_->addOutput(old_out, fusion_->getOutputAlias(old_out).hide_output);
```

Comment:
I'm glad we caught this one.

The previous bug was that completeFusion after segmentation were artificially changing the hide_output from FusionSegmentGuard.

Reply:
alternatively, we should also have used `fusion_` instead of `fusion` here: https://github.com/NVIDIA/Fuser/blob/baea1bb3b935f71cb1dead5737c6e6c572238393/csrc/kernel_cache.cpp#L602

I'll clean up this part of the code in a follow up PR to remove the permutation stuff on IO.

Reply:
added a specific repro for the failing case.